portal/common/lib/page_fetching.bench.ts

```diff
@@ -1,20 +1,132 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { bench, describe, expect } from 'vitest';
+import { describe, bench, expect, vi, beforeAll, beforeEach, afterAll } from 'vitest';
 import { fetchPage } from './page_fetching';
-import { SuiClient, getFullnodeUrl } from '@mysten/sui/client'
-import { NETWORK } from './constants';
+import { SuiClient, SuiObjectData } from '@mysten/sui/client';
 import { SITES_USED_FOR_BENCHING } from './constants';
+import { sha256 } from './crypto';
+import { toHEX } from '@mysten/bcs';
+
+// Mock content and expected hash.
+const mockContent = '<html>Mock Page Content</html>';
+const contentBuffer = Buffer.from(mockContent, 'utf8');
+let expectedHash: string;
+
+// Mock functions.
+const fetchMock = vi.fn();
+const getDynamicFieldObject = vi.fn();
+const getObject = vi.fn();
+const mockClient = {
+    getDynamicFieldObject,
+    getObject,
+} as unknown as SuiClient;
+
+describe('Page fetching with mocked network calls', () => {
+    beforeAll(async () => {
+        // Set up the global fetch mock.
+        globalThis.fetch = fetchMock;
+
+        // Compute expected hash.
+        const decompressed = new Uint8Array(contentBuffer);
+        const hashArray = await sha256(decompressed);
+        expectedHash = toHEX(hashArray);
+
+        // Mock fetch response.
+        fetchMock.mockResolvedValue({
+            ok: true,
+            status: 200,
+            headers: new Headers({
+                'Content-Type': 'text/html',
+            }),
+            arrayBuffer: async () => contentBuffer,
+            text: async () => mockContent,
+            json: async () => ({ message: 'Mock Page Content' }),
+        } as unknown as Response);
+
+        // Mock 'decompress_data'.
+        vi.mock('./decompress_data', () => ({
+            decompressData: vi.fn(async (data: Uint8Array) => data),
+        }));
+
+        // Mock 'bcs_data_parsing'.
+        vi.mock('./bcs_data_parsing', () => ({
+            ResourcePathStruct: {},
+            ResourceStruct: {},
+            DynamicFieldStruct: () => ({
+                parse: vi.fn().mockReturnValue({
+                    value: {
+                        blob_id: '0xresourceBlobId',
+                        path: '/index.html',
+                        content_type: 'text/html',
+                        content_encoding: 'utf8',
```

Comment:
Very soon this will need to be replaced by `headers` because we will accept any kind of http header using a `map<String,String>` so this mock will be deprecated by EOW. 

To avoid double work, I would suggest rebasing from develop as soon as #210 gets merged and then make this change there.